front/lib/models.ts

```diff
@@ -892,6 +893,69 @@ ChatRetrievedDocument.init(
 
 ChatMessage.hasMany(ChatRetrievedDocument);
 
+export class GensTemplate extends Model<
+  InferAttributes<GensTemplate>,
+  InferCreationAttributes<GensTemplate>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+  declare workspaceId: ForeignKey<Workspace["id"]>;
+  declare userId: ForeignKey<User["id"]>;
+  declare instructions: string[];
+  declare name: string;
+  declare visibility: "user" | "workspace";
+  declare color: string;
+  declare sId: string;
+}
+
+GensTemplate.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    instructions: {
+      type: DataTypes.ARRAY(DataTypes.TEXT),
+      allowNull: false,
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    visibility: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    color: {
+      type: DataTypes.STRING,
+    },
+    sId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "gens_template",
+    sequelize: front_sequelize,
+    indexes: [{ fields: ["workspaceId"] }],
```

Comment:
We likely want an index on [workspaceId, sId] which has to be unique
and maybe an index on [workspaceId, userId, visibility]